config.go

```diff
@@ -27,11 +28,48 @@ type repository struct {
 func (r repository) Packages() []string {
 	pkgs := []string{r.Prefix}
 	for _, s := range r.Subs {
-		pkgs = append(pkgs, path.Join(r.Prefix, s))
+		pkgs = append(pkgs, path.Join(r.Prefix, s.Name))
 	}
 	return pkgs
 }
 
+type sub struct {
+	Name   string `json:"name"`
+	Hidden bool   `json:"hidden"`
+}
+
+func (s *sub) UnmarshalJSON(raw []byte) error {
+	var v interface{}
+
+	err := json.Unmarshal(raw, &v)
+	if err != nil {
+		return err
+	}
+
+	switch t := v.(type) {
```

Comment:
>The current solution for unmarshaling a string or map might not be the best, so if there is a best practice, I'm all ears. I'm still trying to find a better solution.

I tried to think of a better solution too, but I'm at a bit of a loss, I don't think there is a great way to implement this. Alternative ideas would be to:

1. Use the `json.Decoder` to manually decode each token using the `Token()` function on the decoder, but it's very tedious.
2. Attempt to `json.Unmarshal` into a `string` first, if it errors, attempt to `json.Unmarshal` into the struct, if it fails, return the error.

I think 2 maybe is similar to what you've got here, but it doesn't load the json into a map.

Reply:
My best idea so far is using [mapstructure](https://github.com/mitchellh/mapstructure) for further decoding the `map[string]interface{}` type into a sub. That way I don't need the extra struct at all.

Reply:
Added mapstructure in a separate commit, but I can easily revert it if you don't want the extra dependency.

Reply:
Ah, interesting. I think the dependency is overkill for this tool. Given it's only two fields of the map is the easiest method we can pull the two fields out manually. What do you think? Otherwise I think attempting to decode as string first then if that fails attempting to decode as a strict would be simple.

Reply:
> Otherwise I think attempting to decode as string first then if that fails attempting to decode as a strict would be simple.

Unfortunately that won't work, because it will call `UnmarshalJSON` on the struct again and again. That's why I added the inline struct. But for now, reading directly from the map should work as well.

Reply:
Updated the PR to decode values from the map manually.